pkg/pool-linear/contracts/aave/AaveLinearPoolFactory.sol

```diff
@@ -135,7 +165,24 @@ contract AaveLinearPoolFactory is
         address actualRebalancerAddress = Create2.deploy(0, rebalancerSalt, rebalancerCreationCode);
         require(expectedRebalancerAddress == actualRebalancerAddress, "Rebalancer deployment failed");
 
+        // Identify the protocolId associated with this pool. We do not require that the protocolId be registered.
+        emit AaveLinearPoolCreated(address(pool), protocolId);
```

Comment:
for my own understanding: can a pool be created with an incorrect / unregistered protocol ID? Is that expected?

Reply:
Yes; my interpretation of the conversation around this was that as long as a protocol (e.g., Beefy) can create pools with the correct id - so that all legit Beefy pools had that id - they weren't concerned about "false positives" (random non-Beefy pools using the same id).

It's all for the UIs: it's just emitting an event, not storing or validating anything. Maybe the "real" protocol IDs should be odd numbers like 42, 327, 8143 rather than 0,1,2, for that reason. Since "non-participating" protocols would probably just pass 0, we wouldn't want 0 to mean "AAVE"

Reply:
Correct, we don't care about pools with the wrong ID. It'll just mean those Pools have bad data.